src/commands/stepfunctions/helpers.ts

```diff
@@ -228,11 +228,13 @@ export type StepType = {
 
 export type ParametersType = {
   'Payload.$'?: string
+  Payload?: string
   FunctionName?: string
   StateMachineArn?: string
   TableName?: string
   Input?: {
     'CONTEXT.$'?: string
+    CONTEXT?: string
```

Comment:
I was wondering if it's possible and if there's a plan to rename to "Payload" for naming consistency between Lambda and SFN cases

Just curious if it can be added down the line

Reply:
I think a better way is to encapsulate only those fields we need into a field, maybe called `DDTracingContext`. Details:

Right now we are passing all the 11 fields from `$$`:
```
{
  "Execution": {
    "Id": "arn:aws:states:sa-east-1:425362996713:express:YimingTestExpressStateMachine2:a5f1a67a-dd92-401e-ae32-d270d68a1741:f2450599-6c08-4717-a91c-e0b41fd340a0",
    "Input": {},
    "Name": "a5f1a67a-dd92-401e-ae32-d270d68a1741",
    "RoleArn": "arn:aws:iam::425362996713:role/service-role/StepFunctions-YimingTestExpressStateMachine2-role-gydji1d2u",
    "StartTime": "2024-09-13T16:43:32.537Z",
    "RedriveCount": 0
  },
  "StateMachine": {
    "Id": "arn:aws:states:sa-east-1:425362996713:stateMachine:YimingTestExpressStateMachine2",
    "Name": "YimingTestExpressStateMachine2"
  },
  "State": {
    "Name": "Lambda Invoke",
    "EnteredTime": "2024-09-13T16:43:33.543Z",
    "RetryCount": 0
  }
}
```
and the downstream is probably using only a small subset of them. Passing too many field will impact user experience when they check the log. For example, their custom fields can be hard to find from the long JSON string.

I'm thinking of creating:
```
"DDTracingContext": {
  "ExecutionId": "arn:aws:states:sa-east-1:425362996713:express:YimingTestExpressStateMachine2:a5f1a67a-dd92-401e-ae32-d270d68a1741:f2450599-6c08-4717-a91c-e0b41fd340a0",
  "StateName": "Lambda Invoke",
  "StateEnteredTime": "2024-09-13T16:43:33.543Z"
}
```
(assuming only these three fields are what we need. We can add more if needed.)
and add it to
- `Payload` of Lambda
- `Parameters.Input` for Step Function

Does this sound good to you?

Reply:
I really like this idea, way cleaner!

Reply:
Just realized another point, I think this change will also bring some performance wins!

Our current implementation merges the entire Execution which means the Input also gets copied and **duplicated**. We currently truncate the actual Input/Output fields to 10K but this can lead to a case where a large Input can live under the Execution at its full length 😨

Reply:
Good suggestion! I think we can use `DDTraceContext` as other repos do (not `tracing`). 
https://github.com/search?q=org%3ADataDog+ddtracecontext&type=code

I think there are a couple of things to consider:

1. We'll need to update ALL tracer layers for this change. Another change to logs-backend should be straightforward.
2. If AWS put more information into the context object and we want to utilize them (e.g. `Execution.RedriveCount` and `Execution.RedriveTime`), we'll need to update the schema in all tracers & logs-backend and ask customers to update their SF definition again. Although new information added probably only happen every couple of years, but it still seems a lot to ask for our customers.
3. When we inject the whole SF context object into payload, the customers may already expect these fields to be in their lambda for a year now. So, this might be a breaking change for few customers. But this is less of a concern IMO.

### SF Context Object
https://docs.aws.amazon.com/step-functions/latest/dg/input-output-contextobject.html#contextobject-format
```
{
    "Execution": {
        "Id": "String",
        "Input": {},
        "Name": "String",
        "RoleArn": "String",
        "StartTime": "Format: ISO 8601",
        "RedriveCount": Number,
        "RedriveTime": "Format: ISO 8601"
    },
    "State": {
        "EnteredTime": "Format: ISO 8601",
        "Name": "String",
        "RetryCount": Number
    },
    "StateMachine": {
        "Id": "String",
        "Name": "String"
    },
    "Task": {
        "Token": "String"
    }
}
```

Reply:
@kimi-p Thanks for the comment!
> I think we can use `DDTraceContext` as other repos do (not `tracing`).
https://github.com/search?q=org%3ADataDog+ddtracecontext&type=code

Is that a different thing? For example, in your link, the `DDTraceContext` in `datadog-lambda-java` contains `traceID`, `parentID` and `samplingPriority`, not `ExecutionId`, `StateName` and `StateEnteredTime` discussed here. If that's a different thing, I think we need to pick a name that's very different from `DDTraceContext` to avoid confusion.

> If AWS put more information into the context object and we want to utilize them (e.g. `Execution.RedriveCount` and `Execution.RedriveTime`), we'll need to update the schema in all tracers & logs-backend and ask customers to update their SF definition again. Although new information added probably only happen every couple of years, it still seems a lot to ask for our customers.

So here's the tradeoff:
|  | Maintainability (for us) | Overhead (for users) | Performance | Debugging experience (for users) |
| - | - | - | - | - |
| Current approach | Good.  | Good (low). No trouble to update the version | Bad. Passes unnecessary fields | Bad. Too many fields. |
| Proposed approach | Bad. need to make tracer keep backward compatibility for a long time until all clients switch to the new version | Bad (high). Need to update the version when asked by us. | Good. Passes less data | Good. Fewer fields. |

Which option would you go for? Is there a dimension that's particularly important?